### PR TITLE
fix(cluster.py): Generate a scylla core dump according to boolean flag

### PR DESCRIPTION
**Pull request by yarongilor**
When run_nodetool command gets an exception it ignored coredump_on_timeout parameter.
	This is fixed and also added an event before the actual node kill starts.
	Refs: #5418

Fix: #5418 

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2551,8 +2551,8 @@ def run_nodetool(self, sub_cmd, args="", options="", timeout=None,
             except Exception as details:  # pylint: disable=broad-except
                 if isinstance(details, RetryableNetworkException):
                     details = details.original
-                if details.__class__.__name__.endswith("CommandTimedOut"):
-                    self.generate_coredump_file()
+                if coredump_on_timeout and details.__class__.__name__.endswith("CommandTimedOut"):
+                    self.generate_coredump_file(node=self.name)
```
Comment (fgelcer) [inline sdcm/cluster.py:2555]: here you should add a message, to override the default message in the `GeneratingCoreDumpEvent` class...
it should be something like, `sending SIGQUIT because command reached timeout` to have much more detailed print before we create these coredumps on timeout

File: sdcm/cluster.py [hunk 2]
```diff
@@ -2448,8 +2448,8 @@ def prepare_files_for_archive(self, fileslist):
                 self.remoter.run('cp -r %s %s' % (f, new_full_path), ignore_status=True)
         return root_dir
 
-    def generate_coredump_file(self, restart_scylla=True):
-        self.log.info('Generate scylla core')
+    def generate_coredump_file(self, restart_scylla=True, node: Any = None):
+        GeneratingCoreDumpEvent(node=node).publish()
```
Comment (fruch) [inline sdcm/cluster.py:2452]: What's wrong with using InformationalEvent

File: sdcm/cluster.py [hunk 3]
```diff
@@ -2448,8 +2448,8 @@ def prepare_files_for_archive(self, fileslist):
                 self.remoter.run('cp -r %s %s' % (f, new_full_path), ignore_status=True)
         return root_dir
 
-    def generate_coredump_file(self, restart_scylla=True):
-        self.log.info('Generate scylla core')
+    def generate_coredump_file(self, restart_scylla=True, node: Any = None):
```
Comment (fruch) [inline sdcm/cluster.py:2451]: node isn't Any, it should be BaseNode, otherwise there's no meaning for adding type annotations here

File: sdcm/cluster.py (hunk 2 above)
Comment (yarongilor) [reply sdcm/cluster.py:2452]: Roy asked for an error event. and it makes sense since some error must have happened to the node in this scenario,

File: sdcm/cluster.py (hunk 3 above)
Comment (yarongilor) [reply sdcm/cluster.py:2451]: perhaps it better be either a BaseNode or an STR?

File: sdcm/cluster.py (hunk 2 above)
Comment (fruch) [reply sdcm/cluster.py:2452]: first, unittest are failing with this change:
```
ValueError: There is no max severity configured for GeneratingCoreDumpEvent
```

What's wrong with:

```
InformationalEvent(severity=Severity.ERROR)
```

adding an event to sct would normally mean there more then one place it's going to called from
and there more specific information it's going to hold.

in this case you just want an error event raised, with a clear message, I think `InformationalEvent` is enough

Comment (roydahan) [reply sdcm/cluster.py:2452]: InformationalEvent(severity=Severity.ERROR) is good enough.
We don't need a new type of event.

Comment (yarongilor) [reply sdcm/cluster.py:2452]: according to @fruch , he meant using `InfoEvent` (not InformationalEvent). but then calling `InfoEvent(severity=Severity.ERROR)` may not be sufficient in order to override severity.

Comment (yarongilor) [reply sdcm/cluster.py]: fixed

File: sdcm/cluster.py (hunk 3 above)
Comment (yarongilor) [reply sdcm/cluster.py]: fixed

**Comment by yarongilor** [comment]
Comments are addressed. if unit-test will fail then there might be a small update needed for the event severity.

**Review by fruch — APPROVED** [review]
LGTM

File: sdcm/cluster.py [hunk 4]
```diff
@@ -2448,8 +2448,11 @@ def prepare_files_for_archive(self, fileslist):
...
         return root_dir
 
-    def generate_coredump_file(self, restart_scylla=True):
-        self.log.info('Generate scylla core')
+    def generate_coredump_file(self, restart_scylla=True, node_name: str = None, message: str = None):
+        message = message or "Generating a Scylla core dump by SIGQUIT..\n"
+        if node_name:
+            message += f"Node: {node_name}"
+        InfoEvent(severity=Severity.ERROR, message=message)
         self.remoter.run("sudo pkill -f --signal 3 /usr/bin/scylla")
```
Comment (fgelcer) [inline sdcm/cluster.py:2456]: if you are already changing, please us `self.remoter.sudo("pkill -f --signal 3 /usr/bin/scylla")` instead

**Review by fgelcer — CHANGES_REQUESTED** [review]
just a small change required

File: sdcm/cluster.py (hunk 4 above)
Comment (yarongilor) [reply sdcm/cluster.py]: fixed

**Review by fgelcer — APPROVED** [review]
LGTM

File: defaults/severities.yaml [hunk 5]
```diff
@@ -95,7 +95,7 @@ TestFrameworkEvent: CRITICAL
 ElasticsearchEvent: ERROR
 SpotTerminationEvent: CRITICAL
 ScyllaRepoEvent: WARNING
-InfoEvent: NORMAL
+InfoEvent: ERROR
```
Comment (roydahan) [inline defaults/severities.yaml:98]: Why is this needed?

**Review by fruch — APPROVED** [review]
LGTM

File: defaults/severities.yaml (hunk 5 above)
Comment (fruch) [reply defaults/severities.yaml:98]: Since it's using this event with error severity

This file set the maximum severity for each type. (Not that I remember why we need it, i.e. one more way of filtering event I guess)

Comment (roydahan) [reply defaults/severities.yaml:98]: The maximum severity?
Where do we use it?
